functions/New-DbaAvailabilityGroup.ps1

```diff
@@ -545,20 +545,18 @@ function New-DbaAvailabilityGroup {
         if ($ClusterType -eq 'Wsfc') {
             Write-ProgressHelper -StepNumber ($stepCounter++) -Message "Adding endpoint connect permissions"
 
-            foreach ($second in $secondaries) {
-                if ($Pscmdlet.ShouldProcess($Primary, "Adding cluster permissions for availability group named $Name")) {
-                    Write-Message -Level Verbose -Message "WSFC Cluster requires granting [NT AUTHORITY\SYSTEM] a few things. Setting now."
-                    $sql = "GRANT ALTER ANY AVAILABILITY GROUP TO [NT AUTHORITY\SYSTEM]
-                        GRANT CONNECT SQL TO [NT AUTHORITY\SYSTEM]
-                        GRANT VIEW SERVER STATE TO [NT AUTHORITY\SYSTEM]"
-                    try {
-                        $null = $server.Query($sql)
-                        foreach ($second in $secondaries) {
-                            $null = $second.Query($sql)
-                        }
-                    } catch {
-                        Stop-Function -Message "Failure adding cluster service account permissions" -ErrorRecord $_
+            if ($Pscmdlet.ShouldProcess($Primary, "Adding cluster permissions for availability group named $Name")) {
```

Comment:
ahh, i had included the whatif in the secondaries loop but forgot to mention which secondary. i think that'd be useful? Also, should we check if the host is linux? not sure if it's possible for `[NT AUTHORITY\SYSTEM]` to exist there.

Reply:
`if ($ClusterType -eq 'Wsfc') {` - so we are definitly on windows.

Reply:
Maybe the diff is missleading. I just deleted line 548 and the corresponding closing bracket. That changed the indention and so the diff is so bad. You don't see a difference if there is only one element in $secondaries.